
# HG changeset patch
# User Keir Fraser <keir.fraser@citrix.com>
# Date 1275645121 -3600
# Node ID 2bbaf4e06c1d406c7fc5cba3cb53cfdc355aef70
# Parent 754ae5d0077dcf318f41ba22b17209f315a12cee
x86: Fix guest-pointer-array memmove in __pirq_guest_unbind().

Thanks to Alex Zeffert for finding this.

Signed-off-by: Keir Fraser <keir.fraser@citrix.com>
xen-unstable changeset:   21492:96917cf25bf3
xen-unstable date:        Fri May 28 10:54:07 2010 +0100

--- a/xen/arch/x86/irq.c	Fri Jun 04 10:20:33 2010 +0100
+++ b/xen/arch/x86/irq.c	Fri Jun 04 10:52:01 2010 +0100
@@ -736,7 +736,8 @@ static irq_guest_action_t *__pirq_guest_
     for ( i = 0; (i < action->nr_guests) && (action->guest[i] != d); i++ )
         continue;
     BUG_ON(i == action->nr_guests);
-    memmove(&action->guest[i], &action->guest[i+1], IRQ_MAX_GUESTS-i-1);
+    memmove(&action->guest[i], &action->guest[i+1],
+            (action->nr_guests-i-1) * sizeof(action->guest[0]));
     action->nr_guests--;
 
     switch ( action->ack_type )

